File: leetCode/fukun/lc_125.js
const isPalindrome = function (s) {
    let curr1 = 0;
    let curr2 = s.length - 1;
    const p = /[a-zA-Z0-9]/i;
    while (curr2 >= curr1) {
        if (!p.test(s[curr1])) {
            curr1++;
            continue;
        }
        if (!p.test(s[curr2])) {
            curr2--;
            continue;
        }
        if (s[curr1].toLowerCase() !== s[curr2].toLowerCase()) {
            return false;
        }
        curr1++;
        curr2--;
    }
    return true;
};
console.log(isPalindrome("A man, a plan, a canal: Panama"));
console.log(isPalindrome("0P"))

File: leetCode/linovce/lc_125.java
class Solution {
    public boolean isPalindrome(String s) {
        if(s.equals(""))
            return true;
        s = s.toLowerCase();
        for(int i=0,j=s.length()-1;i<j;i++,j--){
            while(!(s.charAt(i)>='a'&&s.charAt(i)<='z'||s.charAt(i)>='0'&&s.charAt(i)<='9')&&i<j){
                i++;
            }
            while(!(s.charAt(j)>='a'&&s.charAt(j)<='z'||s.charAt(j)>='0'&&s.charAt(j)<='9')&&i<j){
                j--;
            }
            if(i<j){
                if(!(s.charAt(i)==s.charAt(j)))
                    return false;
            }
            
        }
        
        return true;
    }
}